src/xtgeo/xyz/_xyz.py

```diff
@@ -1,53 +1,53 @@
 # -*- coding: utf-8 -*-
-"""XTGeo xyz module (base class)"""
-from collections import OrderedDict
-from copy import deepcopy
-import pathlib
+"""XTGeo XYZ module (abstract base class)"""
+from abc import ABC, abstractmethod
 
 import pandas as pd
-
-import xtgeo
-from xtgeo.common import XTGeoDialog, XTGDescription
-from xtgeo.xyz import _xyz_io
-from xtgeo.xyz import _xyz_roxapi
+from xtgeo.common import XTGDescription, XTGeoDialog
+from xtgeo.xyz import _xyz_oper
 
 xtg = XTGeoDialog()
 logger = xtg.functionlogger(__name__)
 
 
-class XYZ:
-    """Base class for Points and Polygons in XTGeo."""
-
-    def __init__(self, *args, **kwargs):
-        """Initiate instance"""
-
-        self._df = None
-        self._ispolygons = False
-        self._xname = "X_UTME"
-        self._yname = "Y_UTMN"
-        self._zname = "Z_TVDSS"
-        self._pname = "POLY_ID"
-        self._mname = "M_MDEPTH"
-        self._filesrc = None
-        # other attributes name: type, where type is
-        # ~ ('str', 'int', 'float', 'bool')
-        self._attrs = OrderedDict()
-
-        if len(args) == 1:
-            if isinstance(args[0], (str, pathlib.Path)):
-                # make instance from file import
-                pfile = args[0]
-                logger.info("Instance from file")
-                fformat = kwargs.get("fformat", "guess")
-                self.from_file(pfile, fformat=fformat)
-
-            if isinstance(args[0], list):
-                # make instance from a list of 3 or 4 tuples
-                plist = args[0]
-                logger.info("Instance from list")
-                self.from_list(plist)
-
-        logger.info("XYZ Instance initiated (base class) ID %s", id(self))
+class XYZ(ABC):
+    """Abstract base class for XYZ objects, i.e. Points and Polygons in XTGeo.
+
+    The XYZ base class has common methods and properties for Points and Polygons. The
+    underlying data storage is a Pandas dataframe with minimal 3 (Points) or 4
+    (Polygons) columns, where the two first represent X and Y coordinates.
+
+    The third column is a number, which may represent the depth, thickness, or other
+    property. For Polygons, there is a 4'th column which is an integer representing
+    poly-line ID, which is handled in the Polygons class. Similarly, Points and Polygons
+    can have additional columns called `attributes`.
+
+    Note:
+        You cannot use the XYZ class directly. Use the :class:`Points` or
+        :class:`Polygons` classes!
+    """
+
+    def __init__(
+        self,
+        xname: str = "X_UTME",
+        yname: str = "Y_UTMN",
+        zname: str = "Z_TVDSS",
+    ):
+        """Concrete initialisation for base class _XYZ."""
+        self._xname = xname
+        self._yname = yname
+        self._zname = zname
+
```

Comment:
Any reason for not including `self._name` and `self._filesrc`? They are common to `Points` and `Polygons`

Reply:
Since they were not exposed as a public API, it seemed best to stick to the minimum of what is exposed instead. Note that `Points` never had an exposed name or filesrc, and `Polygons` never exposed a filesrc.